atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameUnmapper.java

```diff
@@ -51,7 +51,7 @@ public String getShortTableNameFromMappingTable(
             return unmappingCache.get(fullTableName, () -> {
                 SqlConnection conn = null;
                 try {
-                    conn = connectionSupplier.getNewUnsharedConnection();
+                    conn = connectionSupplier.get();
```

Comment:
Any code path that grabs a second connection while already holding one is prone to deadlocks. Consider the following pseudocode:
```
Connection c1 = pool.getConnection();
Connection c2 = pool.getConnection();
doSomething();
c2.close();
c1.close();
```
Now let's say your connection pool has N connection and you have N parallel requests (threads) going through the code path above concurrently. Let's say that each thread executed the first line simultaneously and grabbed the first connection from the pool. Since there are N threads, they exhausted the entire pool and there are no connections left. In order to make progress, each of them needs to grab another connection at line 2. But the pool is empty and we won't return any connections to it until one of the threads is done (but it can't be done because it needs a connection first). Hence we have a deadlock.

Now, I didn't run into the deadlock in `OracleTableNameUnmapper`, but I ran into one in `TableValueStyleCache` while running tests in our internal product with a reduced connection pool size. The code path goes through `AbstractKeyValueService.truncateTables()` which executes a bunch of tasks in parallel, one per table that we are truncating. Then each task goes through `OracleDdlTable.truncate()`. First it calls `conns.get()` which checks out the first connection, and then `truncateOverflowTableIfItExists() -> valueStyleCache.getTableType()` which checks out the second connection and deadlocks. So this commit changes `getTableType()` to reuse the existing connection instead of trying to check out a new one.

As a part of this change, I'm proposing to run the Postgres DbKvs test suite with the connection pool size of 1 to make sure we don't use more than one connection at once anywhere, and hence eliminate deadlocks like this. It would be great if we could do this to our internal Oracle tests, too.